packages/diagrams/frontend/sirius-components-diagrams/src/palette/tool-section/ToolSection.types.ts

```diff
@@ -14,5 +14,6 @@ import { GQLTool, GQLToolSection } from '../ContextualPalette.types';
 
 export interface ToolSectionProps {
   toolSection: GQLToolSection;
+  defaultToolId: string | undefined;
```

Comment:
Minor nitpick, either you want to explicitly handle nothing as an explicit value and in this case use `string | null` but if you want to provide an optional prop on a component you should probably use `defaultToolId?: string`. With this, you do not have to specify `<ToolSection defaultToolId={undefined} />` which is not very sexy.

Most of the time, I prefer that we handle explicitly null values but for the props of a component, an optional prop is perfectly fine. It works the same for simple boolean-based props for example to be able to have something like this:

```
<Something />
<Something />
<Something disabled />
<Something />
```

And not

```
<Something disabled={false} />
<Something disabled={false} />
<Something disabled={true} />
<Something disabled={false} />
```